src/Sulu/Bundle/AdminBundle/Resources/js/containers/DeleteDependantResourcesDialog/DeleteDependantResourcesDialog.js

```diff
@@ -0,0 +1,262 @@
+// @flow
+import React from 'react';
+import {observer} from 'mobx-react';
+import {toJS, action, computed, observable} from 'mobx';
+import equals from 'fast-deep-equal';
+import Dialog from '../../components/Dialog';
+import {type SnackbarType} from '../../components/Snackbar';
+import ProgressBar from '../../components/ProgressBar';
+import ResourceRequester from '../../services/ResourceRequester';
+import RequestPromise from '../../services/Requester/RequestPromise';
+import {translate} from '../../utils';
+import styles from './deleteDependantResourcesDialogStyles.scss';
+import type {Resource, DependantResourcesData, DependantResourceBatches} from '../../types';
+
+type Props = {
+    dependantResourcesData: DependantResourcesData,
+    onCancel?: () => void,
+    onClose?: () => void,
+    onError?: (error: any) => void,
+    onFinish?: () => void,
+    requestOptions?: {[string]: any} | null,
+}
```

Comment:
do we need to add a `open` prop that is passed to the `Dialog` to keep the animation?

Reply:
If we handle the open state from outside, after deletion is finished, we would have to pass `open={false}` and an empty object as `dependentResourcesData` prop which causes the `DeleteDependantResourcesDialog` to reset it's internal state because of the `componentDidUpdate` ... That would lead to a change of the `Close Dialog` button text to `Cancel` just when the dialog is closed ... So I think it's not worth it to handle it from outside. Also, when deletion is triggered inside the form, the view is closed anyways, then it doesn't matter if the dialog closes nicely or is just gone. WDYT?